### PR TITLE
Screw you CSS

### PR DESCRIPTION
**Pull request by bamader**
## Background
This PR swaps out the old nav bars (primary and secondary) with a single unified navigation bar in the upper right corner of the app.

## GitHub Issue
- https://github.com/ctoec/data-collection/issues/1349

## Associated PRs
- https://github.com/ctoec/component-library/pull/158

## Validation Plan
- [ ] Verify that the logout option does not appear while a user is not logged in
After a user is logged in:
- [ ] Hovering over an element displays blue background styling
- [ ] Roster link takes user to roster page
- [ ] Roster nav item is underlined while user is on roster page
- [ ] Clicking the Resource nav item opens dropdown, does not take user to a page
- [ ] Template nav item works
- [ ] Data Requirements nav item works
- [ ] Privacy policy nav item works
- [ ] Clicking Help opens that submenu, does not take user to a page
- [ ] How to guides opens another tab and directs user to help guide
- [ ] Support nav item takes user to support page
- [ ] Log out nav item works

## Automated Testing
- [ ] All unit tests are passing.

## Additional Context
The CSS for this PR SUCKED because apparently the confluence of USWDS styling mixed with Carbon styling means nothing likes to get overridden (hence all the `!important` tags in the scss file). I was able to achieve _almost_ all of the functionality in the ticket, but there are two caveats I have:
1. You'll notice two different colors of blue used, one for highlighting and one in the submenu. This is because I wasn't sure which shade of blue was requested in the ticket. It _looks_ like the blue that happens on hover is closest to the ticket, but the blue in the dropdown submenus is a "skylight branded" blue so idk if we want to use that. I don't really care either way, just putting the options there.
2. I could not for the life of me get the dropdown menus to open on hover rather than on click. The header component is relatively poorly documented usage wise in the carbon docs (because it's all part of the UI Shell usage pattern rather than its own page), and between those docs and lots of github issues / googling I couldn't find or figure out any way to make the buttons trigger on hover. I think the Carbon `HeaderMenuItem`s might just only work on click 😕 . Hopefully that's not a big blocker, but also open to other ideas on solving it.

### CODE REVIEW

File: client/src/containers/App/App.scss [hunk 1]
```diff
@@ -3,3 +3,43 @@ nav.usa-nav {
     list-style: none;
   }
 }
+
+.bx--text-truncate--end {
```
Comment (JordanGuinn) [inline client/src/containers/App/App.scss:7]: Think it'd be worth putting these styles alongside the corresponding component they now apply to in `component-library`?

**Review by JordanGuinn — APPROVED** [review]

File: client/src/containers/App/App.tsx [hunk 2]
```diff
@@ -1,55 +1,57 @@
...
-    {
-      text: 'Help',
-      type: 'secondary',
-      path: '/help',
-    },
-  ];
+   // TODO: When we tackle the admin nav bar, this will change because
+  // We'll probably use a pattern of a single top right nav bar,
+  // so the admin things may eventually migrate there.
+  // TLDR: The comment below (which is from the original page) is not true.
```
Comment (jlhogan) [inline client/src/containers/App/App.tsx:16]: wait so..... what's the current state? can comments that are not true be removed?

Comment (bamader) [reply client/src/containers/App/App.tsx:16]: The current state still has the admin view seeing two navs, one in the top right and one in the upper left of the page part. Discussion on the ticket said that one of the incoming next steps was to standardize the nav view so that admin users had the same thing as other users--everyone sees one nav bar in the top right. So I just put this todo there to document that we'll eventually move the admin nav items to where normal users' nav items are. Can delete if it's not helpful.

File: client/src/containers/App/App.scss (hunk 1 above)
Comment (bamader) [reply client/src/containers/App/App.scss:7]: As in, move this particular css class into the component library instead of defining it here?

Comment (JordanGuinn) [reply client/src/containers/App/App.scss:7]: Yep!  Well really all these classes, so that in case we end up using this kinda header somewhere else in the application, it's already styled how we'd expect.

File: client/src/containers/App/App.tsx (hunk 2 above)
Comment (JordanGuinn) [reply client/src/containers/App/App.tsx:16]: +1 for comment removal - I'm a little confused by them, honestly 😕

**Comment by bamader** [comment]
Comments removed, and CSS styles shifted over to component library. This is ready for re-review (or just continued review).

**Comment by JordanGuinn** [comment]
## Validation Plan
- [ ] Confirm current roster navbar has been removed entirely
- [ ] Confirm admin navbar is still in place for OEC admins
- [ ] Verify that the logout option does not appear while a user is not logged in
- [ ] Hovering over an element displays blue background styling
- [ ] Roster link takes user to roster page
- [ ] Roster nav item is underlined while user is on roster page
- [ ] Roster nav item is NOT underlined while user is away from roster page
- [ ] Clicking the Resource nav item opens dropdown, does not take user to a page
   - [ ] Template nav takes user to template page
   - [ ] Data Requirements nav takes user to data requirements page
   - [ ] Privacy policy nav takes user to privacy policy page
- [ ] Clicking Help opens that submenu, does not take user to a page
   - [ ] How to guides opens another tab and directs user to help guide
   - [ ] Support nav item takes user to support page
   - [ ] Log out nav logs user out

**Comment by JordanGuinn** [comment]
After my validation pass, I'd say _functionally_ everything is looking as it should!  Just found one minor issue and a few UI tweaks that should probably be addressed at least in part:

- Only the `Log Out` button is displayed for admin users; think it should probably be everything except the `Roster` link?
- `Roster` button is highlighted on landing page, when the user is not logged in
- Navbar buttons and `ECE Reporter` icon in left-hand corner aren't aligned vertically
- Navbar elements are pretty short, compared to the `ECE Reporter` icon
- Nav item dropdown options aren't highlighted when you hover over them

**Comment by JordanGuinn** [comment]
Re-ran my validation steps, and all looks to be addressed!  The last two bullet points captured above are still outstanding, but in my opinion not problematic enough to be considered blockers.  We'll have a chance during regression testing of this whole roster refactor epic to cull out things like that, anyway.

TL;DR I'd say this is ready for merge!
